# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/InsertTextTransaction.h
# Commit: 1466d7d61102
# Full Hash: 1466d7d6110279cf167ab8b1567be693df5d7c3c
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2020-03-03 09:50:30
# Regressor Bug: 1618089
# File Overlap Count: 2
# Description:
#   Bug 1618089 - part 7: Make some related methods use `EditorDOMPointInText` for arguments or result r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D64338
# ==============================================================================

diff -r 079f4b05ead2 -r 1466d7d61102 editor/libeditor/InsertTextTransaction.h
--- a/editor/libeditor/InsertTextTransaction.h	Tue Mar 03 02:19:59 2020 +0000
+++ b/editor/libeditor/InsertTextTransaction.h	Tue Mar 03 01:15:38 2020 +0000
@@ -6,7 +6,9 @@
 #ifndef InsertTextTransaction_h
 #define InsertTextTransaction_h
 
-#include "mozilla/EditTransactionBase.h"   // base class
+#include "mozilla/EditTransactionBase.h"  // base class
+
+#include "mozilla/EditorDOMPoint.h"        // EditorDOMPointInText
 #include "nsCycleCollectionParticipant.h"  // various macros
 #include "nsID.h"                          // NS_DECLARE_STATIC_IID_ACCESSOR
 #include "nsISupportsImpl.h"               // NS_DECL_ISUPPORTS_INHERITED
@@ -36,22 +38,20 @@
 class InsertTextTransaction final : public EditTransactionBase {
  protected:
   InsertTextTransaction(EditorBase& aEditorBase,
-                        const nsAString& aStringToInsert, dom::Text& aTextNode,
-                        uint32_t aOffset);
+                        const nsAString& aStringToInsert,
+                        const EditorDOMPointInText& aPointToInsert);
 
  public:
   /**
    * Creates new InsertTextTransaction instance.  This never returns nullptr.
    *
    * @param aEditorBase     The editor which manages the transaction.
-   * @param aTextNode       The text content node to be inserted
-   *                        aStringToInsert.
-   * @param aOffset         The offset in aTextNode to do the insertion.
+   * @param aPointToInsert  The insertion point.
    * @param aStringToInsert The new string to insert.
    */
   static already_AddRefed<InsertTextTransaction> Create(
       EditorBase& aEditorBase, const nsAString& aStringToInsert,
-      dom::Text& aTextNode, uint32_t aOffset);
+      const EditorDOMPointInText& aPointToInsert);
 
   NS_DECLARE_STATIC_IID_ACCESSOR(NS_INSERTTEXTTXN_IID)
 